src/client/index.tsx

```diff
@@ -24,59 +24,80 @@ const notebookApi = acquireNotebookRendererApi(JupyterNotebookRenderer);
+    metadata?: OldNotebookCellOutputMetadata;
+}
+interface OldNotebookOutputEventParams {
+    element: HTMLElement;
+    outputId: string;
+    output: OldNotebookOutput;
+    mimeType: string;
+}
+
 /**
  * Called from renderer to render output.
  * This will be exposed as a public method on window for renderer to render output.
  */
 function renderOutput(request: NotebookOutputEventParams) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const mimeString = request.mime || (request as any).mimeType;
     try {
-        console.error('request', request);
+        console.log('request', request);
         const output = convertVSCodeOutputToExecutResultOrDisplayData(request);
-        console.log(`Rendering mimeType ${request.mimeType}`, output);
-        console.error('request output', output);
+        console.log(`Rendering mimeType ${mimeString}`, output);
+        console.log('request output', output);
 
-        ReactDOM.render(React.createElement(CellOutput, { mimeType: request.mimeType, output }, null), request.element);
+        ReactDOM.render(React.createElement(CellOutput, { mimeType: mimeString, output }, null), request.element);
     } catch (ex) {
-        console.error(`Failed to render mime type ${request.mimeType}`, ex);
+        console.error(`Failed to render mime type ${mimeString}`, ex);
     }
 }
 
 function convertVSCodeOutputToExecutResultOrDisplayData(
     request: NotebookOutputEventParams
 ): nbformat.IExecuteResult | nbformat.IDisplayData {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const metadata: Record<string, any> = {};
-    // Send metadata only for the mimeType we are interested in.
-    const customMetadata = request.output.metadata?.custom;
-    if (customMetadata) {
-        // Support for Old API
-        if (customMetadata[request.mimeType]) {
-            metadata[request.mimeType] = customMetadata[request.mimeType];
-        }
-        if (customMetadata.needs_background) {
-            metadata.needs_background = customMetadata.needs_background;
-        }
-        if (customMetadata.unconfined) {
-            metadata.unconfined = customMetadata.unconfined;
-        }
+    if ('mime' in request) {
+        // New API
+        return {
+            data: {
+                [request.mime]: request.value
+            },
+            metadata: request.metadata || {},
+            execution_count: null,
+            output_type: request.metadata?.outputType || 'execute_result'
+        };
     } else {
-        // New API.
+        // Old API
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const outputMetadata = request.output.metadata as Record<string, any> | undefined;
-        if (outputMetadata && outputMetadata[request.mimeType] && outputMetadata[request.mimeType].metadata) {
+        const metadata: Record<string, any> = {};
+
+        const oldRequest = (request as unknown) as OldNotebookOutputEventParams;
```

Comment:
Is casting to `unknown` better practice?

Reply:
That was what I was reading. (Full confession, I didn't actually know this beforehand, and looked it up after a hint from a TS compiler warning). Here is Anders commenting on the addition of unknown. Basically a typesafe `any` so that you can use it safely as an intermediate with casting.
https://github.com/Microsoft/TypeScript/pull/24439

Reply:
TIL, thanks :)